hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/metadata/SchemaOneChunkInfoListCodec.java

```diff
@@ -43,38 +39,13 @@
  * future, callers should be aware that the associated chunk information may not
  * always be present.
  */
-public final class SchemaOneChunkInfoListCodec implements Codec<ChunkInfoList> {
-
-  private static final Codec<ChunkInfoList> INSTANCE =
-      new SchemaOneChunkInfoListCodec();
-
+public final class SchemaOneChunkInfoListCodec {
+  /** @return {@link ChunkInfoList#getCodec()}. */
   public static Codec<ChunkInfoList> get() {
-    return INSTANCE;
+    return ChunkInfoList.getCodec();
   }
 
   private SchemaOneChunkInfoListCodec() {
     // singleton
   }
-
-  @Override
-  public byte[] toPersistedFormat(ChunkInfoList chunkList) {
-    return chunkList.getProtoBufMessage().toByteArray();
-  }
-
-  @Override
-  public ChunkInfoList fromPersistedFormat(byte[] rawData) throws IOException {
-    try {
-      return ChunkInfoList.getFromProtoBuf(
-              ContainerProtos.ChunkInfoList.parseFrom(rawData));
-    } catch (InvalidProtocolBufferException ex) {
-      throw new IOException("Invalid chunk information. " +
-              "This data may have been written using datanode " +
-              "schema version one, which did not save chunk information.", ex);
-    }
```

Comment:
When implementing HDDS-8559 I kept this separate, mostly duplicated class due to the difference in this exception handling.  Code compiles fine because `InvalidProtocolBufferException` is an `IOException`, but are you sure it has the same behavior after the change?  (I'm not sure it's covered by tests.)

Reply:
To play safe, let me revert `SchemaOneChunkInfoListCodec` and keep it unchanged.

Reply:
Different behavior is possible if some code catches both `InvalidProtocolBufferException` and `IOException` and handles them separately.  The only case where I found such code is:

https://github.com/apache/ozone/blob/f80f0561bf15819ae78b1c681a3f154d160f363a/hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java#L369-L373

I'm not sure it's even related to `SchemaOneChunkInfoListCodec`, but anyway it doesn't seem to be a significant difference, so I think we should be OK with the change.